irohad/execution/impl/command_executor.cpp

```diff
@@ -1054,7 +1054,37 @@ namespace iroha {
       const shared_model::interface::DetachRole &command,
       ametsuchi::WsvQuery &queries,
       const shared_model::interface::types::AccountIdType &creator_account_id) {
-    return {};
+    auto command_name = "DetachRole";
+    auto role_permissions = queries.getRolePermissions(command.roleName());
+    auto account_roles = queries.getAccountRoles(creator_account_id);
+    auto subject_account_roles = queries.getAccountRoles(command.accountId());
+
+    if (not role_permissions or not account_roles
+        or not subject_account_roles) {
+      return makeCommandError(
+          "is valid command validation failed: unable to query roles info",
```

Comment:
Occurrences of `"is valid command validation failed"` looks and sounds very strange to me. Please consider rephrasing.